.github/workflows/required-tests.yml

```diff
@@ -0,0 +1,211 @@
+          run: tar -xzf pki-build-${{ matrix.os }}/pki-rpms.tar.gz
+
+        - name: Install required packages
+          run: docker exec -i ${CONTAINER} dnf install -y 389-ds-base-legacy-tools findutils dnf-plugins-core sudo wget 389-ds-base
+
+        - name: Install PKI packages
+          run: docker exec -i ${CONTAINER} bash -c "find ${PKIDIR} -name '*.rpm' -and -not -name '*debuginfo*' | xargs dnf -y install"
+        
+        #- name: Enable PKI COPR repo
+        #  run: docker exec -i ${CONTAINER} dnf copr enable -y ${COPR_REPO}
+
+        - name: Install DS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ds-create.sh
+
+        - name: Install CA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ca-create.sh
+
+        - name: Install KRA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/kra-create.sh
+
+        - name: Install OCSP
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ocsp-create.sh
+
+        - name: Install TKS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tks-create.sh
+
+        - name: Install TPS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tps-create.sh
+
+        - name: Remove TPS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tps-remove.sh
+
+        - name: Remove TKS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tks-remove.sh
+
+        - name: Remove OCSP
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ocsp-remove.sh
+
+        - name: Remove KRA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/kra-remove.sh
+
+        - name: Remove CA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ca-remove.sh
+
+        - name: Remove DS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ds-remove.sh
+
+        - name: Gather journalctl logs
+          run: docker exec -i ${CONTAINER} bash -c "journalctl -u pki-tomcatd@pkitest > /var/log/pki/pki-journalctl.log"
+
+        - name: Compress logs
+          if: always()
+          run: docker exec -i ${CONTAINER} bash -c "tar -czf ${PKIDIR}/pki-logs.tar.gz /var/log/pki"
+
+        - name: Upload Logs
+          if: always()
+          uses: actions/upload-artifact@v1
+          with:
+            name: pki-logs-${{ matrix.os }}
+            path: pki-logs.tar.gz
+
+    # Test job
+    ipa-tests:
+      # This job depends on the 'build' job and waits till it completes.
+      # This job needs container to be started manually, as Github provided
+      # container **does not** use /usr/bin/init as its ENTRYPOINT.
+      name: Test IPA
+      needs: build
+      runs-on: ubuntu-latest
+      env:
+        CONTAINER: pkitest
+        BUILDDIR: /tmp/workdir
+        PKIDIR: /tmp/workdir/pki
+        LOGS: ${GITHUB_WORKSPACE}/logs.txt
+        COPR_REPO: "@pki/master"
+        test_set: "test_caacl_plugin.py test_caacl_profile_enforcement.py test_cert_plugin.py test_certprofile_plugin.py test_ca_plugin.py test_vault_plugin.py"
```

Comment:
Since the build is now only done once, maybe we can split some of the IPA tests to run concurrently to reduce the execution time.

Reply:
Yes, that's possible too \o/ May be in a future PR?